dataset/src/main/scala/frameless/Job.scala

```diff
@@ -22,8 +22,17 @@ sealed abstract class Job[A](implicit sc: SparkContext) { self =>
       }
     }
   }
+
+  def map[B](fn: A => B): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run())
+  }
+
+  def flatMap[B](fn: A => Job[B]): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run()).run()
+  }
 }
 
+
 object Job {
   def apply[A](a: => A)(implicit sc: SparkContext): Job[A] = new Job[A] {
```

Comment:
Since we are touching this file, shouldn't we move to using Spark Session instead of the rather less relevant SparkContext? @kanterov ?